sway-core/src/language/ty/declaration/function.rs

```diff
@@ -182,12 +180,95 @@ impl MaterializeConstGenerics for TyFunctionDecl {
     }
 }
 
+/// Rename cosnt generics when the name inside the struct/enum does not match
+/// the name in the impl items.
+fn rename_const_generics(
+    engines: &Engines,
+    impl_self_or_trait: &TyImplSelfOrTrait,
+    m: &mut TyFunctionDecl,
```

Comment:
`m` hardly carries any meaning as a name. Also, in the online review, it's veey hard to spot it in the code. Can we have a more descriptive name here? Just `impl_fn` would IMO already be a better choice.